File: array/121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ret = 0
        minx = sys.maxsize
        
        for i in prices:
            minx = minx if minx < i else i
            ret = ret if ret > (i - minx) else (i - minx)
        return ret

File: array/121.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) {
            return 0;
        }
        int ret = 0;
        int min = prices[0];
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] < min) {
                min = prices[i];
                continue;
            }
            if (prices[i] - min > ret) {
                ret = prices[i] - min;
            }
        }
        return ret;
    }
}
